src/introduction.tex

```diff
@@ -0,0 +1,3 @@
+\chapter{Introduction}\label{ch:introduction}
+In the introduction, you give context for your research, introduce your research question, and outline your work as well as motivate it.
+It is also common to add a list of contributions, i.e., bullet points on what you contribute to the research area.
```

Comment:
The introduction is one of the most important sections. Some things that could be explained here:

- the "zooming in" aspect of an introduction: start from "computer science" zoom into the specific subfield (and motivate it), zoom into the topic at hand, etc.
- the goal of the introduction is to most importantly answer the reader's questions "why should I care?", "what is the problem?", "how will solving it improve the state of the art?", "why should I trust the author on this?", "what are the specific contributions?"
- for a BSc / MSc thesis it should be approachable by a non-expert (computer scientist, but not a PL researcher).

Reply:
though it might be controversial, I used something akin to the following:
```
The introduction should address the following key questions, ideally in the order stated below:
\begin{enumerate}
\item What is the nature of the problem this thesis investigates?
\item What are the current common approaches for solving this problem?
\item How does this thesis address the problem?
\item What are the results? What improved?
\item What can the reader expect in the individual chapters of the thesis?
\end{enumerate}

The recommended length for the introduction is between 1 and 4 pages. For longer introductions, consider dividing the content into unnumbered subsections with descriptive headings. However, avoid extensive discussions of specific topics here; detailed content like definitions, theorems, methods, and data belongs in subsequent chapters.
```